src/support/generation.c

```diff
@@ -126,11 +125,9 @@ __wt_gen_drain(WT_SESSION_IMPL *session, int which, uint64_t generation)
      * session count. That way, no matter what sessions come or go, we'll check the slots for all of
      * the sessions that could have been active when we started our check.
      */
-    WT_ORDERED_READ(session_cnt, conn->session_cnt);
-    for (minutes = 0, pause_cnt = 0, s = conn->sessions, i = 0; i < session_cnt; ++s, ++i) {
-        if (!s->active)
-            continue;
-
+    minutes = 0;
+    pause_cnt = 0;
+    WT_SESSION_FOREACH_BEGIN(s, conn) {
         for (;;) {
             /* Ensure we only read the value once. */
             WT_ORDERED_READ(v, s->generations[which]);
```

Comment:
Not part of this change, this will become a `WT_READ_ONCE` in a different ticket.